### PR TITLE
Adopt SOMA naming scheme

### PR DESCRIPTION
**Pull request by aaronwolen**
From [NEWS.md](https://github.com/TileDB-Inc/tiledbsc/blob/aaronwolen/sc-17952/adopt-soma-naming-scheme/NEWS.md):

This changes the names of the 2 top-level classes in the tiledbsc package to follow new nomenclature adopted by the [single-cell data model specification](https://github.com/single-cell-data/matrix-api/blob/main/specification.md), which was implemented [here](https://github.com/single-cell-data/matrix-api/pull/28). You can read more about the rationale for this change [here](https://github.com/single-cell-data/matrix-api/issues/11#issuecomment-1109975498).

## New class names

- `SCGroup` is replaced by `SOMA` (stack of matrices, annotated)
- `SCDataset` is replaced by `SOMACollection`

There are no functional changes to either class. `SOMA` is a drop-in replacement for `SCGroup` and `SOMACollection` is a drop-in replacement for `SCDataset`. However, with the new names two of `SOMACollection`'s methods have changed accordingly:

- the `scgroups` field is now `somas`
- `scgroup_uris()` is now `soma_uris()`

To ease the transition, the `SCDataset` and `SCGroup` classes are still available as aliases for `SOMACollection` and `SOMA`, respectively. However, they have been deprecated and will be removed in the future.

### CODE REVIEW

File: vignettes/introduction.Rmd [hunk 1]
```diff
@@ -41,98 +41,98 @@ data("pbmc_small", package = "SeuratObject")
...
+fs::dir_tree(soma$uri, recurse = FALSE)
 ```
 
 ## Accessing individual components
 
-Any of the underlying TileDB arrays can be accessed directly from an `sc_dataset` object by navigating its internal classes.
+Any of the underlying TileDB arrays can be accessed directly from a `SOMACollection` object by navigating its internal classes.
 
-As an example, let's access the cell-level metadata. Recall from the [`matrix-api spec`][matrix-api-spec] that cell-level metadata is stored in the `obs` array of an `SCGroup`. Therefore, we must first access a specific `SCGroup` within the `SCDataset`'s `scgroups` slot. Let's generate a list of the available `scgroups`:
+As an example, let's access the cell-level metadata. Recall from the [`matrix-api spec`][matrix-api-spec] that cell-level metadata is stored in the `obs` array of an `SOMA`. Therefore, we must first access a specific `SOMA` within the `SOMACollection`'s `somas` slot. Let's generate a list of the available `somas`:
```
Comment (johnkerl) [inline vignettes/introduction.Rmd:112]: an -> a

**Review by johnkerl — APPROVED** [review]
lgtm, with nits

**Review by eddelbuettel — APPROVED** [review]
Looks good and quite comprehensive and as discussed over slack the deprecation notice both needed and 'The Right Thing (TM)' to do.   Thanks also to @johnkerl for using a fine comb on typos and wording consistency.

**Comment by aaronwolen** [comment]
CC @dnadave @kaitlin-procogia @augustine-procogia

**Pull request merged**